### PR TITLE
Migrate to k8s.io/klog from glog.

### PR DESCRIPTION
**Pull request by humblec**
Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

### CODE REVIEW

**Comment by humblec** [comment]
@pohly thanks , I missed it. Added  in latest comment. PTAL .

File: Gopkg.toml [hunk 1]
```diff
@@ -67,6 +67,11 @@
 [[constraint]]
   name = "github.com/kubernetes-csi/csi-lib-utils"
   version = "0.1.0"
+
+[[constraint]]
+  name = "k8s.io/klog"
+  revision = "8139d8cb77af419532b33dfa7dd09fbc5f1d344f"
+
```
Comment (pohly) [inline Gopkg.toml]: Why this constraint?

If there is a reason, explain it in a comment.

Please don't use the `Gopkg.toml` file to lock onto a specific revision. That's what the `Gopk.lock` file is for, with `dep ensure -update` as the mechanism for updating to a new version.

**Comment by humblec** [comment]
> Why is glog not getting removed from vendor?

@pohly I was expecting glog to go away as we have `unused-package=true` in toml. But it didnt, looks like due to `constraint` tag. any way removing it from toml and updating the PR.

> 
> glog and klog cannot co-exist in the same binary, so I'd expect glog to be removed entirely as part of this PR.

Done.

**Comment by humblec** [comment]
@pohly if I remove `glog` , build fails:
```
mkdir -p bin
CGO_ENABLED=0 GOOS=linux go build -a -ldflags '-X main.version=v1.0.0-rc2-70-g1482c63d -extldflags "-static"' -o ./bin/csi-provisioner ./cmd/csi-provisioner
vendor/github.com/kubernetes-sigs/sig-storage-lib-external-provisioner/controller/controller.go:31:2: cannot find package "github.com/golang/glog" in any of:
	/home/travis/gopath/src/github.com/kubernetes-csi/external-provisioner/vendor/github.com/golang/glog (vendor tree)
	/home/travis/.gimme/versions/go1.11.1.linux.amd64/src/github.com/golang/glog (from $GOROOT)
	/home/travis/gopath/src/github.com/golang/glog (from $GOPATH)
make: *** [build-csi-provisioner] Error 1
make: Target `all' not remade because of errors.
```

**Comment by pohly** [comment]
Someone needs to remove glog from github.com/kubernetes-sigs/sig-storage-lib-external-provisioner first, then we can switch to that new version together with removing glog in this PR.

**Comment by pohly** [comment]
And the PR is already pending: https://github.com/kubernetes-sigs/sig-storage-lib-external-provisioner/pull/25

**Comment by jsafrane** [comment]
Replaced by https://github.com/kubernetes-csi/external-provisioner/pull/225
/close

**Pull request closed without merging**